File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Munk/MunkB_Arr.m
% MunkB_Arr: arrival calculation along range / single depth (start at r = 0)

clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite
addpath /home/orodrig/FORdoc/at/Matlab/Plot

disp('Bellhop - Munk profile:')
disp('Arrivals calculation, ASCII  file output')
disp('Geometric hat beams in Cartesian coordinates') 

system('bellhop.exe MunkB_Arr');

[Arr, Pos] = read_arrivals_asc( 'MunkB_Arr.arr', 18 );

A         = Arr.A;
delay     = Arr.delay;
Narr      = Arr.Narr; 
NumTopBnc = Arr.NumTopBnc;
NumBotBnc = Arr.NumBotBnc;
RcvrAngle = Arr.RcvrAngle;
SrcAngle  = Arr.SrcAngle;

zs     = Pos.s.depth;
rarray = Pos.r.range;
zarray = Pos.r.depth;
    
Nsd = length( zs );
Nrd = length( zarray );
Nrr = length( rarray );

figure(1), hold on
for i = 2:Nrr

tau =  delay(i,1:Narr(i)); 
amp = abs( A(i,1:Narr(i)) );
rangei = rarray(i)*ones(1,Narr(i));
stem3(tau,rangei,amp)
end 
box on, grid on
hold off
view(22,30)
xlabel('Travel time (s)')
ylabel('Range (in m)')
zlabel('Amplitude')
title('Bellhop - Munk profile')

disp('done.')

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Munk/MunkB_Arr.py
#==================================================================
#  
#  Bellhop - MunkB_Arr: arrival calculation along range / single depth (start at r = 0)
#  Faro, Dom Dez 18 20:57:38 WET 2016
#  Written by Tordar
#  
#==================================================================

from os import *
import sys
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from numpy import *
from scipy.io import *
from pylab import *
from mpl_toolkits.mplot3d import Axes3D
from read_arrivals_asc import *

print("Bellhop - Munk profile:")
print("Arrivals calculation, ASCII  file output")
print("Geometric hat beams in Cartesian coordinates") 

system("bellhop.exe MunkB_Arr")

Arr, Pos = read_arrivals_asc( 'MunkB_Arr.arr', 18 )

Narr   = int_( squeeze( Arr['Narr'] ) )
delay  = real( squeeze( Arr['delay'] ) )
A      = abs( squeeze( Arr['A'] ) )
rarray = squeeze( Pos['r_range'] )

Nrr = rarray.size

fig = figure()
ax = fig.gca(projection='3d')

for i in range(1,Nrr):
    tau = delay[i,0:Narr[i]]
    amp = abs( A[i,0:Narr[i]] )
    rangei = rarray[i]*ones(Narr[i])
    for j in range(Narr[i]):
        ax.plot([tau[j],tau[j]], [rangei[j],rangei[j]], [0.0,amp[j]],'b')    
        ax.plot([tau[j],tau[j]], [rangei[j],rangei[j]], [0.0,amp[j]],'bo')
xlabel('Travel time (s)')
ylabel('Range (in m)')
title('Bellhop - Munk profile')
show()

print('done.')
